tests/no-cross-package-imports.js

```diff
@@ -0,0 +1,110 @@
+		},
+		{
+			code: 'import getSubNodes from \'./utils/getsubnodes\';',
+			filename: path.win32.join( 'C:', 'Workspace', 'ckeditor', 'ckeditor5', 'packages', 'ckeditor5-watchdog', 'src', 'plugin.js' )
+		},
+		// Cross package imports in allowed package
+		{
+			code: 'import { toArray } from \'ckeditor5/src/utils\';',
+			filename: path.win32.join( 'C:', 'Workspace', 'ckeditor', 'ckeditor5', 'packages', 'ckeditor5-core', 'src', 'plugin.js' )
+		},
+		// Other file formats
+		{
+			code: 'import { toArray } from \'ckeditor5/src/utils\';',
+			filename: path.win32.join( 'C:', 'Workspace', 'ckeditor', 'ckeditor5', 'packages', 'ckeditor5-core', 'src', 'plugin.css' )
+		},
+		{
+			code: 'import { toArray } from \'ckeditor5/src/utils\';',
+			filename: path.win32.join( 'C:', 'Workspace', 'ckeditor', 'ckeditor5', 'packages', 'ckeditor5-core', 'src', 'plugin.svg' )
+		},
+
+		/**
+		 * Unix style path
+		 */
+
+		// Non-cross package imports in disallowed package
+		{
+			code: 'import Watchdog from \'./watchdog\';',
+			filename: path.posix.join( 'C:', 'Workspace', 'ckeditor', 'ckeditor5', 'packages', 'ckeditor5-watchdog', 'src', 'plugin.js' )
+		},
+		{
+			code: 'import EditorWatchdog from \'./editorwatchdog\';',
+			filename: path.posix.join( 'C:', 'Workspace', 'ckeditor', 'ckeditor5', 'packages', 'ckeditor5-watchdog', 'src', 'plugin.js' )
+		},
+		{
+			code: 'import areConnectedThroughProperties from \'./utils/areconnectedthroughproperties\';',
+			filename: path.posix.join( 'C:', 'Workspace', 'ckeditor', 'ckeditor5', 'packages', 'ckeditor5-watchdog', 'src', 'plugin.js' )
+		},
+		{
+			code: 'import getSubNodes from \'./utils/getsubnodes\';',
+			filename: path.posix.join( 'C:', 'Workspace', 'ckeditor', 'ckeditor5', 'packages', 'ckeditor5-watchdog', 'src', 'plugin.js' )
+		},
+		// Cross package imports in allowed package
+		{
+			code: 'import { toArray } from \'ckeditor5/src/utils\';',
+			filename: path.posix.join( 'C:', 'Workspace', 'ckeditor', 'ckeditor5', 'packages', 'ckeditor5-core', 'src', 'plugin.js' )
+		},
+		// Other file formats
+		{
+			code: 'import { toArray } from \'ckeditor5/src/utils\';',
+			filename: path.posix.join( 'C:', 'Workspace', 'ckeditor', 'ckeditor5', 'packages', 'ckeditor5-core', 'src', 'plugin.css' )
+		},
+		{
+			code: 'import { toArray } from \'ckeditor5/src/utils\';',
+			filename: path.posix.join( 'C:', 'Workspace', 'ckeditor', 'ckeditor5', 'packages', 'ckeditor5-core', 'src', 'plugin.svg' )
+		}
+	],
+	invalid: [
+		{
+			code: 'import { toArray } from \'ckeditor5/src/utils\';',
+			filename: path.win32.join( 'C:', 'Workspace', 'ckeditor', 'ckeditor5', 'packages', 'ckeditor5-watchdog', 'src', 'plugin.js' ),
+			errors: [ importError ]
+		},
+		{
+			code: 'import { toArray } from \'ckeditor5/src/utils\';',
+			filename: path.posix.join( 'C:', 'Workspace', 'ckeditor', 'ckeditor5', 'packages', 'ckeditor5-watchdog', 'src', 'plugin.js' ),
+			errors: [ importError ]
+		}
```

Comment:
I'd expect more invalid scenarios:

* `import toArray from '@ckeditor/ckeditor5-utils/src/toarray';`
* `import { toArray } from '@ckeditor/ckeditor5-utils';`

For both paths, Windows, and Unix.